nixos/modules/services/web-apps/mattermost.nix

```diff
@@ -77,6 +157,16 @@ in
         '';
       };
 
+      preferNixConfig = mkOption {
```

Comment:
I'm not sure if I like this. It seems like this just overrides the config at server startup so you can still configure things differently in the web UI but you will suddenly loose settings then. I understand that you actively need to enable it but I don't know if we really need/want this except maybe If you are using this.

Reply:
You used to lose all the settings if you disabled mutable configs entirely. This provides an intermediate option.